### PR TITLE
Constructors on Polkadot can be non-payable

### PR DESCRIPTION
**Pull request by xermicus**
We use the `seal1` `instantiate` runtime API, and with this runtime host function there is no need to send the minimum balance on instantiation (the comment in the docs is not correct).

Hence, constructors on Polkadot can and should actually be non-payble

### CODE REVIEW

**Review by seanyoung — CHANGES_REQUESTED** [review]
Looks good.

I noticed that the dispatch code no longer has a special case where if all functions are non-payable, then there is a single value transfer check which is both less code and less gas used in the error case. This could be utilized in the constructor case too.

File: integration/polkadot/package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "Integration tests with Solang and Polkadot",
   "main": "index.js",
   "scripts": {
-    "test": "tsc; ts-mocha -t 20000 --exit *.spec.ts",
+    "test": "tsc; ts-mocha -t 20000 --exit runtime_errors.spec.ts",
```
[the remark this answers is not in the document]
Comment (xermicus) [reply integration/polkadot/package.json]: Oh I forgot to push my last commit to enable them again, thanks!

**Comment by xermicus** [comment]
> there is a single value transfer check 

Very good point thanks. You are right. Either all constructors are payable or all are non-payable (this is enforced in sema).

**Comment by xermicus** [comment]
Hmm so I thought about it.
- I have not yet seen a real world example with multiple constructors. And if there is a single nonpayable constructor, a single transferred value check will be in the code and exectued anyways. So no real gains here.
- It will make the code size a bit smaller if there are multiple contract functions and all of them are nonpayable. However this is unrelated to this PR.

@seanyoung WDYT about opening an issue so I can just implement it in another PR?

**Review by seanyoung — APPROVED** [review]

**Comment by seanyoung** [comment]
> @seanyoung WDYT about opening an issue so I can just implement it in another PR?

Note that this functionality was in the dispatch code in emit for functions.

**Review by LucasSte — COMMENTED** [review]
Is there a test case for when a payable constructor receives no value?

**Comment by xermicus** [comment]
@LucasSte `polkadot_tests::value.rs` contains two tests that explicitly test payable constructor receiving 0 value either because the user explicit set zero value or it was not set respectively.

**Review by LucasSte — APPROVED** [review]

**Pull request merged**